hibernate-core/src/test/java/org/hibernate/test/schemaupdate/SchemaMigrationToOutputScriptTest.java

```diff
@@ -0,0 +1,179 @@
+		sessionFactory.close();
+	}
+
+	private void createServiceRegistryAndMetadata(String append) {
+		final StandardServiceRegistryBuilder standardServiceRegistryBuilder = new StandardServiceRegistryBuilder()
+				.applySetting( Environment.FORMAT_SQL, "false" )
+				.applySetting( Environment.HBM2DDL_SCRIPTS_ACTION, "update" )
+				.applySetting( AvailableSettings.HBM2DDL_SCRIPTS_CREATE_TARGET, output.getAbsolutePath() );
+
+		if ( append != null ) {
+			standardServiceRegistryBuilder.applySetting( AvailableSettings.HBM2DDL_SCRIPTS_CREATE_APPEND, append );
+		}
+
+		serviceRegistry = standardServiceRegistryBuilder.build();
+
+		metadata = (MetadataImplementor) new MetadataSources( serviceRegistry )
+				.addAnnotatedClass( MyEntity.class )
+				.addAnnotatedClass( MySecondEntity.class )
+				.buildMetadata();
+		metadata.validate();
+	}
+
+	@After
+	public void tearDown() {
+		ServiceRegistryBuilder.destroy( serviceRegistry );
+		serviceRegistry = new StandardServiceRegistryBuilder().applySetting( AvailableSettings.HBM2DDL_AUTO, "drop" )
+				.build();
+
+		metadata = (MetadataImplementor) new MetadataSources( serviceRegistry )
+				.addAnnotatedClass( MyEntity.class )
+				.buildMetadata();
+		final SessionFactory sessionFactory = metadata.buildSessionFactory();
+		sessionFactory.close();
+	}
+
+	@Test
+	public void testAppendModeFalse() throws Exception {
+		createServiceRegistryAndMetadata( "false" );
+		HashMap properties = new HashMap<>();
+		properties.putAll( serviceRegistry.getService( ConfigurationService.class ).getSettings() );
+
+		SchemaManagementToolCoordinator.process(
+				metadata,
+				serviceRegistry,
+				properties,
+				null
+		);
+		List<String> commands = Files.readAllLines( output.toPath() );
+		assertThat( commands.size(), is( 1 ) );
+		assertTrue( commands.get( 0 ).contains( createTableMySecondEntity ) );
+	}
+
+	@Test
+	public void testAppendModeTrue() throws Exception {
+		createServiceRegistryAndMetadata( "true" );
+		HashMap properties = new HashMap<>();
+		properties.putAll( serviceRegistry.getService( ConfigurationService.class ).getSettings() );
+
+		SchemaManagementToolCoordinator.process(
+				metadata,
+				serviceRegistry,
+				properties,
+				null
+		);
+		List<String> commands = Files.readAllLines( output.toPath() );
+		assertThat( commands.size(), is( 10 ) );
+		assertTrue( commands.get( 9 ).contains( createTableMySecondEntity ) );
+	}
+
+	@Test
+	public void testDefaultAppendMode() throws Exception {
+		createServiceRegistryAndMetadata( null );
+		HashMap properties = new HashMap<>();
+		properties.putAll( serviceRegistry.getService( ConfigurationService.class ).getSettings() );
+
+		SchemaManagementToolCoordinator.process(
+				metadata,
+				serviceRegistry,
+				properties,
+				null
+		);
+		List<String> commands = Files.readAllLines( output.toPath() );
+		assertThat( commands.size(), is( 10 ) );
+		assertTrue( commands.get( 9 ).contains( createTableMySecondEntity ) );
```

Comment:
Not defect, just sharing idea. We might well use 'StringContains' hamcrest matcher residing in the same package as 'Is':
```
assertThat( commands.get( 9 ), containsString( createTableMySecondEntity ) );
```

Reply:
Thanks @NathanQingyangXu , the `containsString` is definitely better than the `assertTrue` and `contains`!